File: lecture_notes/11 review/between.py

# Extract substring between prefix and suffix

def get_value_after(str, prefix):
  prefix_start = str.find(prefix)
  if prefix_start == -1:
    return ''
  resultStart = prefix_start + len(prefix)
  return str[resultStart:]


def get_value_before(str, suffix):
  suffix_start = str.find(suffix)
  if suffix_start == -1:
    return ''
  return str[0:suffix_start]


def get_value_between(str, prefix, suffix = None):  
  str = get_value_after(str, prefix)
  if suffix == None:
    return str
  str = get_value_before(str, suffix)
  return str

result = get_value_between('Hello <username> and bye!', '<', '>')
print(result)

File: lecture_notes/11 review/between.js
'use strict';
// Extract substring between prefix and suffix

const getValueAfter = (str, prefix) => {
  const prefixStart = str.indexOf(prefix);
  if (prefixStart === -1) return '';
  const resultStart = prefixStart + prefix.length;
  return str.substring(resultStart);
}

const getValueBefore = (str, suffix) => {
  const suffixStart = str.indexOf(suffix);
  if (suffixStart === -1) return '';
  return str.substring(0, suffixStart);
}

const getValueBetween = (str, prefix, suffix) => {
  str = getValueAfter(str, prefix);
  if (!suffix) return str;
  str = getValueBefore(str, suffix);
  return str
}

const result = getValueBetween('Hello <username> and bye!', '<', '>');
console.log(result);

